File: yunshu_music/android/app/src/main/java/top/itning/yunshu_music/service/MusicPlayMode.java
package top.itning.yunshu_music.service;

/**
 * @author itning
 * @since 2021/10/12 15:05
 */
public enum MusicPlayMode {
    SEQUENCE,
    RANDOMLY,
    LOOP,
    ;

    public static MusicPlayMode getNext(MusicPlayMode nowMode) {
        switch (nowMode) {
            case SEQUENCE:
                return MusicPlayMode.RANDOMLY;
            case RANDOMLY:
                return MusicPlayMode.LOOP;
            case LOOP:
            default:
                return MusicPlayMode.SEQUENCE;
        }
    }
}

File: yunshu_music/macos/Runner/MusicPlayMode.swift
import Foundation

enum MusicPlayMode:String {
    case SEQUENCE,RANDOMLY,LOOP
    
    static func getNext(_ nowMode:MusicPlayMode)->MusicPlayMode{
        switch nowMode {
        case .SEQUENCE:
            return MusicPlayMode.RANDOMLY
        case .RANDOMLY:
            return MusicPlayMode.LOOP
        case .LOOP:
            return MusicPlayMode.SEQUENCE
        }
    }
    
    static func valueOf(_ value:String)->MusicPlayMode{
        if value == MusicPlayMode.LOOP.rawValue{
            return .LOOP
        }
        if value == MusicPlayMode.SEQUENCE.rawValue{
            return .SEQUENCE
        }
        if value == MusicPlayMode.RANDOMLY.rawValue{
            return .RANDOMLY
        }
        return .SEQUENCE
    }
}
